src/wallet/rpcwallet.cpp

```diff
@@ -2405,26 +2405,26 @@ static UniValue loadwallet(const JSONRPCRequest& request)
             + HelpExampleCli("loadwallet", "\"test.dat\"")
             + HelpExampleRpc("loadwallet", "\"test.dat\"")
         );
-    std::string wallet_file = request.params[0].get_str();
+
+    WalletLocation location(request.params[0].get_str());
     std::string error;
 
-    fs::path wallet_path = fs::absolute(wallet_file, GetWalletDir());
-    if (fs::symlink_status(wallet_path).type() == fs::file_not_found) {
-        throw JSONRPCError(RPC_WALLET_NOT_FOUND, "Wallet " + wallet_file + " not found.");
-    } else if (fs::is_directory(wallet_path)) {
+    if (!location.Exists()) {
+        throw JSONRPCError(RPC_WALLET_NOT_FOUND, "Wallet " + location.GetName() + " not found.");
+    } else if (fs::is_directory(location.GetPath())) {
```

Comment:
Possible future improvement: It might be nice to add a `WalletLocation` method to hold directory checking logic below. Maybe `WalletLocation::HasData`, or `WalletLocation::IsEmpty`, or something like that. After #14552 (aef19be5851ceb3b324fd80756060a00aad8e61c) this could also use `SplitWalletPath`.

Reply:
Is it bad to add `WalletLocation::GetEnvironmentPath()` and `WalletLocation::GetDatabasePath()`?

Reply:
> Is it bad to add WalletLocation::GetEnvironmentPath() and WalletLocation::GetDatabasePath()?

It's not terrible, but I think it'd be better not to have berkeley stuff baked into the WalletLocation interface. Maybe a wallet location doesn't need to be a btree file and environment directory. Maybe it could be some other type of database path or connection string.

I think it would be an improvement if this code were more like:

```c++
if (!location.Exists()) {
  throw RPCError("Specified wallet path doesn't exist. Call createwallet to create a wallet at this path.");
} else if (!location.HasData()) {
  throw RPCError("Not loading wallet. Specified path doesn't does not appear to be a valid wallet.");
}
```

But I think more ideally BerkeleyDatabase would have a `bool create` option that that `loadwallet` would set `false` and `createwallet` would set `true`, so error checking at this level could be removed entirely, and shared with the wallet tool from #13926, for example.